### PR TITLE
Large refactor of Int7Bh to support wbtrv32

### PR DESCRIPTION
**Pull request by paladine**
Added unit tests for testing all the methods
Fixed several bugs
Implemented logical currency in GetDirect which we didn't support before. Not sure if the recovery tool depended on that functionality, but it works now.

### CODE REVIEW

File: MBBSEmu.Tests/DOS/Interrupts/Int7Bh_Tests.cs [hunk 1]
```diff
@@ -0,0 +1,978 @@
...
+  public class Int7Bh_Tests : TestBase, IDisposable
+  {
+    private readonly string[] _runtimeFiles = { "MBBSEMU.DB" };
+
+    private readonly ICpuRegisters _registers = new CpuRegisters();
+    private readonly FakeClock _fakeClock = new FakeClock();
+    private readonly ServiceResolver _serviceResolver;
+    private readonly IMemoryCore _memory;
+    private readonly Int7Bh _int7B;
+    private string _modulePath;
```
Comment (enusbaum) [inline MBBSEmu.Tests/DOS/Interrupts/Int7Bh_Tests.cs:43]: These can be readonly as well

File: MBBSEmu.Tests/DOS/Interrupts/Int7Bh_Tests.cs [hunk 2]
```diff
@@ -0,0 +1,978 @@
...
+        foreach (var file in _runtimeFiles)
+        {
+            File.WriteAllBytes(Path.Combine(_modulePath, file), resourceManager.GetResource($"MBBSEmu.Tests.Assets.{file}").ToArray());
+        }
+    }
+
+    [Fact]
+    public void InvalidInterface()
+    {
+      DOSInterruptBtrieveCommand command = new DOSInterruptBtrieveCommand()
```
Comment (enusbaum) [inline MBBSEmu.Tests/DOS/Interrupts/Int7Bh_Tests.cs:87]: var on these

File: MBBSEmu/Btrieve/BtrieveFileProcessor.cs [hunk 3]
```diff
@@ -449,7 +451,7 @@ private BtrieveRecord GetAndCacheBtrieveRecord(uint id, SqliteDataReader reader,
         }
 
         /// <summary>
-        ///     Returns the Record at the specified Offset, while also updating Position to match.
+        ///     Returns the Record at the specified physical ffset, while also updating Position to match.
```
Comment (enusbaum) [inline MBBSEmu/Btrieve/BtrieveFileProcessor.cs:454]: `offset`

File: MBBSEmu/DOS/Interrupts/Int7Bh.cs [hunk 4]
```diff
@@ -12,48 +12,6 @@
...
-        TransactionIsActive = 37,
-        /* Btrieve version 5.x returns this status code
-if you attempt to perform a Step, Update, or Delete operation on a
-key-only file or a Get operation on a data only file */
-        OperationNotAllowed = 41,
-        AccessDenied = 46,
-        InvalidInterface = 53,
-    }
-
     enum BtrieveOpenMode : short
```
Comment (enusbaum) [inline MBBSEmu/DOS/Interrupts/Int7Bh.cs:15]: VS is saying this is unused

File: MBBSEmu/DOS/Interrupts/Int7Bh.cs [hunk 5]
```diff
@@ -135,36 +120,55 @@ public void Dispose()
...
             // return status code back to program
             _memory.SetWord(command.status_code_pointer_segment, command.status_code_pointer_offset, (ushort)status);
             // and update data_buffer_length if it was updated in Handle
-            _memory.SetWord(_registers.DS, (ushort)(_registers.DX + 4), command.data_buffer_length);
+            _memory.SetWord(_registers.DS, (ushort)(_registers.DX + 4), data_buffer_length);
+        }
+
+        private (BtrieveError, ushort) Handle(DOSInterruptBtrieveCommand command)
+        {
+            BtrieveCommand actualCommand = new BtrieveCommand() {
```
Comment (enusbaum) [inline MBBSEmu/DOS/Interrupts/Int7Bh.cs:140]: var

**Review by enusbaum — CHANGES_REQUESTED** [review]

File: MBBSEmu/DOS/Interrupts/Int7Bh.cs (hunk 4 above)
Comment (paladine) [reply MBBSEmu/DOS/Interrupts/Int7Bh.cs:15]: not really used currently, but I'd still like to keep the structure definition around.

**Review by enusbaum — APPROVED** [review]

**Pull request merged**